src/components/panel/panel.spec.js

```diff
@@ -105,6 +108,62 @@ describe('$mdPanel', function() {
     expect(PANEL_EL).not.toExist();
   });
 
+  it('should call onDomAdded if provided when adding the panel to the DOM', function() {
+    var onDomAdded = jasmine.createSpy();
+    var animationHookConfig = angular.extend({'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+    panelRef = $mdPanel.create(animationHookConfig);
+
+    panelRef.attachOnly().then(function() {
+      expect(onDomAdded).toHaveBeenCalled();
+      expect(PANEL_EL).toExist();
+    });
+  });
+
+  it('should call onOpenComplete if provided after adding the panel to the DOM and animating', function() {
+    var onOpenComplete = jasmine.createSpy();
+    var animationHookConfig = angular.extend({'onOpenComplete': onOpenComplete}, DEFAULT_CONFIG);
+
+    panelRef = $mdPanel.create(animationHookConfig);
+
+    openPanel();
+
+    expect(onOpenComplete).toHaveBeenCalled();
+    expect(PANEL_EL).toExist();
+  });
+
+  it('should call onRemoving if provided when removing the panel from the DOM', function() {
+    var onRemoving = jasmine.createSpy();
+    var onDomRemoved = jasmine.createSpy();
+    var animationHookConfig = angular.extend({'onRemoving': onRemoving, 'onDomRemoved': onDomRemoved}, DEFAULT_CONFIG);
+
+    panelRef = $mdPanel.create(animationHookConfig);
+
+    openPanel();
+
+    closePanel().then(function() {
+      expect(onDomRemoved).not.toHaveBeenCalled();
+      expect(onRemoving).toHaveBeenCalled();
+      expect(PANEL_EL).toExist();
+    });
+
+  });
+
+  it('should call onDomRemoved if provided when removing the panel from the DOM', function() {
```

Comment:
For each of these callbacks, there should be 3 tests:

1. That the user can provide a function and it gets called. (Which is the group you wrote here.)
2. That the user can return a promise that resolves and everything continues appropiately.
3. That the user can return a promise that rejects and stops the panel. (For instance, the dev wants to show an "Are you sure you want to delete this?" dialog, and if the user responds "no", the panel does not close.)

Please add the other two types of tests. Also, since there will be 12 tests, that seems like enough for a general describe block.